types/address.go

```diff
@@ -662,30 +490,23 @@ func (ca ConsAddress) Format(s fmt.State, verb rune) {
 	case 'p':
 		s.Write([]byte(fmt.Sprintf("%p", ca)))
 	default:
-		s.Write([]byte(fmt.Sprintf("%X", []byte(ca))))
+		s.Write([]byte(fmt.Sprintf("%X", ca.Bytes())))
 	}
 }
 
 // ----------------------------------------------------------------------------
 // auxiliary
 // ----------------------------------------------------------------------------
 
-var errBech32EmptyAddress = errors.New("decoding Bech32 address failed: must provide a non empty address")
-
-// GetFromBech32 decodes a bytestring from a Bech32 encoded string.
-func GetFromBech32(bech32str, prefix string) ([]byte, error) {
-	if len(bech32str) == 0 {
-		return nil, errBech32EmptyAddress
-	}
+var errHexEmptyAddress = errors.New("decoding hex address failed: must provide a non empty address")
 
-	hrp, bz, err := bech32.DecodeAndConvert(bech32str)
-	if err != nil {
-		return nil, err
+// GetFromHex decodes a bytestring from a hex encoded string.
+func GetFromHex(hexStr, _ string) ([]byte, error) {
```

Comment:
This function could just call `addressBytesFromHexString`. This means that this hunk would be easier to maintain (simpler to understand)

Reply:
Agree

Reply:
Done in latest commit

Reply:
Resolved